src/Requests.php

```diff
@@ -224,49 +224,78 @@ public static function add_transport($transport) {
 	}
 
 	/**
-	 * Get a working transport
+	 * Get the FQCN for a working transport.
 	 *
-	 * @throws \WpOrg\Requests\Exception If no valid transport is found (`notransport`)
-	 * @return \WpOrg\Requests\Transport
+	 * @param array<string, bool> $capabilities Associative array of capabilities to test against, i.e. `['<capability>' => true]`.
+	 * @return string FQCN of the transport to use, or an empty string if no transport was
+	 *                found which provided the requested capabilities.
 	 */
-	protected static function get_transport($capabilities = array()) {
-		// Caching code, don't bother testing coverage
+	protected static function get_transport_class($capabilities = array()) {
+		// Caching code, don't bother testing coverage.
 		// @codeCoverageIgnoreStart
-		// array of capabilities as a string to be used as an array key
+		// Array of capabilities as a string to be used as an array key.
 		ksort($capabilities);
 		$cap_string = serialize($capabilities);
 
-		// Don't search for a transport if it's already been done for these $capabilities
+		// Don't search for a transport if it's already been done for these $capabilities.
 		if (isset(self::$transport[$cap_string]) && self::$transport[$cap_string] !== null) {
-			$class = self::$transport[$cap_string];
-			return new $class();
+			return self::$transport[$cap_string];
 		}
 		// @codeCoverageIgnoreEnd
 
 		if (empty(self::$transports)) {
 			self::$transports = self::DEFAULT_TRANSPORTS;
 		}
 
-		// Find us a working transport
+		// Find us a working transport.
 		foreach (self::$transports as $class) {
 			if (!class_exists($class)) {
 				continue;
 			}
 
-			$result = call_user_func(array($class, 'test'), $capabilities);
+			$result = $class::test($capabilities);
 			if ($result) {
```

Comment:
Let's make this check type specific ?
```suggestion
			if ($result === true) {
```